### PR TITLE
feat(saved-searches): Hide 'Set as Default' button when on default search

### PR DESCRIPTION
**Pull request by malwilley**
Closes https://github.com/getsentry/team-coreui/issues/23

### CODE REVIEW

File: static/app/views/issueList/issueListSetAsDefault.spec.tsx [hunk 1]
```diff
@@ -72,4 +81,30 @@ describe('IssueListSetAsDefault', () => {
...
+    MockApiClient.addMockResponse({
+      url: '/organizations/org-slug/searches/',
+      body: [TestStubs.Search({isPinned: false, query: 'browser:firefox'})],
+    });
+
+    render(<IssueListSetAsDefault {...defaultProps} query="is:unresolved" />, {
+      organization,
+    });
+
+    expect(screen.queryByRole('button')).not.toBeInTheDocument();
```
Comment (evanpurkhiser) [inline static/app/views/issueList/issueListSetAsDefault.spec.tsx:95]: @scttcper pointed out that you can use expect(container).toBeEmptyDOMElement();` for this type of thing too

File: static/app/views/issueList/issueListSetAsDefault.spec.tsx [hunk 2]
```diff
@@ -34,34 +42,35 @@ describe('IssueListSetAsDefault', () => {
...
-        {...defaultProps}
-        savedSearch={TestStubs.Search({isPinned: true})}
-      />
-    );
+    render(<IssueListSetAsDefault {...defaultProps} />, {organization});
 
-    userEvent.click(screen.getByRole('button', {name: /remove default/i}));
+    await waitFor(() => {
+      userEvent.click(screen.getByRole('button', {name: /remove default/i}));
+    });
```
Comment (evanpurkhiser) [inline static/app/views/issueList/issueListSetAsDefault.spec.tsx:73]: Why do you need this? does can't you just use something like 

```suggestion
    userEvent.click(await screen.findByRole('button', {name: /remove default/i}));
```

Comment (evanpurkhiser) [reply static/app/views/issueList/issueListSetAsDefault.spec.tsx:73]: Or do we have to wait for the button to become enabled first or something

Comment (malwilley) [reply static/app/views/issueList/issueListSetAsDefault.spec.tsx:73]: Nope, the way you wrote it works and is better, don't remember why I had it like this

File: static/app/views/issueList/issueListSetAsDefault.tsx [hunk 3]
```diff
@@ -1,33 +1,50 @@
...
   sort: string;
 }
 
+const usePinnedSearch = () => {
+  const organization = useOrganization();
+  const {data: savedSearches} = useFetchSavedSearchesForOrg(
+    {orgSlug: organization.slug},
+    {
+      notifyOnChangeProps: ['data'],
+      enabled: organization.features.includes('issue-list-saved-searches-v2'),
```
Comment (evanpurkhiser) [inline static/app/views/issueList/issueListSetAsDefault.tsx:29]: Didn't we kill this feature flag?

Comment (malwilley) [reply static/app/views/issueList/issueListSetAsDefault.tsx:29]: Yeah I need to rebase

File: static/app/views/issueList/issueListSetAsDefault.spec.tsx (hunk 1 above)
Comment (malwilley) [reply static/app/views/issueList/issueListSetAsDefault.spec.tsx:95]: That is a more accurate way to test this, but I've been avoiding using that because if we make `render` render modals (or anything else) by default that line will fail

File: static/app/views/issueList/overview.tsx [hunk 4]
```diff
@@ -65,6 +65,7 @@ import IssueListFilters from './filters';
 import GroupListBody from './groupListBody';
 import IssueListHeader from './header';
 import {
+  DEFAULT_ISSUE_STREAM_SORT,
```
Comment (evanpurkhiser) [inline static/app/views/issueList/overview.tsx:63]: These are mostly unrelated changes right?

**Review by evanpurkhiser — APPROVED** [review]

File: static/app/views/issueList/issueListSetAsDefault.spec.tsx (hunk 1 above)
Comment (evanpurkhiser) [reply static/app/views/issueList/issueListSetAsDefault.spec.tsx:91]: Ah hmmm. Thoughts @scttcper

File: static/app/views/issueList/overview.tsx (hunk 4 above)
Comment (malwilley) [reply static/app/views/issueList/overview.tsx:63]: I needed to know what the default sort was in `IssueListSetAsDefault`, so I moved that constant to the utils file

Comment (evanpurkhiser) [reply static/app/views/issueList/overview.tsx:63]: Oh I see cool

File: static/app/views/issueList/issueListSetAsDefault.spec.tsx (hunk 1 above)
Comment (scttcper) [reply static/app/views/issueList/issueListSetAsDefault.spec.tsx:91]: i only use it when the entire container is returning null
